src/MyPlot/MyPlot.php

```diff
@@ -273,9 +272,10 @@ public function teleportPlayerToPlot(Player $player, Plot $plot, bool $center =
 		$plotLevel = $this->getLevelSettings($plot->levelName);
 		if($plotLevel === null)
 			return false;
-		$pos = $this->getPlotPosition($plot);
-		$plotSize = $plotLevel->plotSize;
-		$pos->add(floor($plotSize / 2), 1.5, -1);
+		$pos    = $this->getPlotPosition($plot);
+		$pos->x += floor($plotLevel->plotSize / 2);
+		$pos->y += 1;
+		$pos->z -= 1;
```

Comment:
What a great change. The actual fix just would've been:
Changing
`$pos->add(floor($plotSize / 2), 1.5, -1);`
To
`$pos = $pos->add(floor($plotSize / 2), 1.5, -1);`

Reply:
I changed this because I want the player to teleport to the plot level if it’s different from the player’s current level. My change also fixed the bad cordinates bug

Reply:
Well, you code does not affect the level at all. Did you notice that?

Reply:
Mine doesn’t, but yours would have provided a Vector3 object instead of a Position object; hence I used my fix instead to use the Position with the Level instance included. The Vector3::add() function returns a Vector3 object, and there is no override in Position to return a Position object, so the Fox I provided works